### ISSUE TITLE
DocValues - Are there plans to read/display them in Luke ?

### ISSUE THREAD

**Issue by kiwiclive** [DmitryKey/luke#24]
This is not really an issue, more of a question :-)

With the introduction of updatable fields in Lucene, by the use of DocValues, are there plans to allow Luke to decode and display the content of DocValue fields ?

Thanks,
Clive

**Comment by DmitryKey** [comment]
@kiwiclive luke currently marks the field with D for DocValues. I have began working on decoding the DV from a field. Question: what sort of information are you looking for in particular for DV?

**Comment by DmitryKey** [comment]
@kiwiclive would be great, if you gave your pointers feature-wise. So far I was able to extract the doc ids as document values. Not sure it is what you are looking for.

**Comment by grossws** [comment]
I think, docValues type should be extracted (like SORTED/SORTED_SET/etc), and tokens for current document (there are ords in SORTED_SET usually, so ord -> term mapping would be usefull).

**Comment by DmitryKey** [comment]
@grossws makes sense. I guess, the question is more on presentation layer, i.e. what and how to compactly show this. The can be cases when tens of thousands of documents share the same docValue.

but then on top of this -- what will this info be useful for in terms of interacting with the index (except for mere checking the sanity of values), any ideas?

**Comment by mocobeta** [comment]
FYI, improved doc values supports (decode & display doc values data, save them to local files, etc.) were added in Pivot version Luke.
Here are details and screenshots: https://github.com/DmitryKey/luke/pull/53

Please try pivot-luke branch if you like it. Feedbacks are welcome!
